nodestream/pipeline/transformers/expand_json_field.py

```diff
@@ -19,7 +19,12 @@ def __init__(self, path: List[str]) -> None:
     async def transform_record(self, record: JsonLikeDocument):
         item = record
         for path_segment in self.path[:-1]:
-            record = item[path_segment]
+            item = item.get(path_segment, None)
```

Comment:
Not sure why `item = record` above is needed. The interlaced use of `record` and `item` is extremely confusing.

For example, this line could be:
`path_segment= record.get(path_segment, None)`

then below:
`last_segment_value = path_segment.get(last_segment, None)

Reply:
I guess that won't work... let me think what else can be done